pkgs/by-name/pu/purescm/package.nix

```diff
@@ -0,0 +1,45 @@
+{ lib
+, buildNpmPackage
+, fetchNpmDeps
+, testers
+}:
+
+let
+  package = buildNpmPackage rec {
+    pname = "purescm";
+    version = "1.0.0";
+
+    src = ./.;
+
+    dontNpmBuild = true;
+
+    npmDeps = fetchNpmDeps {
+      inherit src;
+      hash = "sha256-rSm40Q9A5xfMQIsH6QQaCcemU9LnLmvGdLmII4nApsI=";
+      postPatch = ''
+        cp ${./package-lock.json} package-lock.json
+      '';
+    };
+
+    installPhase = ''
+      mkdir -p $out/share/${pname}
+      cp -r node_modules/ $out/share/${pname}
+      mkdir -p $out/bin
+      ln -s $out/share/${pname}/node_modules/.bin/purescm $out/bin
+
+    '';
+
+    passthru.tests = if lib.versionOlder "1.0.1" version then { } else {
+      version = testers.testVersion { inherit package; };
```

Comment:
This should be done using finalAttrs

Reply:
see https://github.com/NixOS/nixpkgs/blob/cac03116a08561c8b9f8ae040918f28d4fb81483/doc/languages-frameworks/pkg-config.section.md#writing-packages-providing-pkg-config-modules-pkg-config-writing-packages as an example